events/wc_void_events.txt

```diff
@@ -2371,3 +2371,158 @@ narrative_event = {
 		clr_character_flag = found_cthraxxi_bad_end_flag
 	}
 }
+
+### Tekahn's Bargain ###
+# An agent of the Twilight's Hammer visits the Neferset and offers to lift the Curse of Flesh
+narrative_event = {
+	id = WCVOD.10100
+	title = EVTTITLE_WCVOD_10100
+	desc = EVTDESC_WCVOD_10100
+	picture = GFX_evt_cthun_priestess
+	border = GFX_event_narrative_frame_intrigue
+
+	is_triggered_only = yes
+
+	trigger = {
+		has_landed_title = c_neferset	# Must hold the city of Neferset
+		culture = tolvir	# Must be of tol'vir culture
+		is_shadow_priest_class_trigger = yes	# Only shadow priests can be contacted by the entity
+		has_global_flag = uldum_has_opened_flag		# Uldum must not be isolated
+		NOT = { has_global_flag = dark_pharaohs_bargain_flag }	# This is here so event can't fire twice
+	}
+
+	immediate = {
+		set_global_flag = dark_pharaohs_bargain_flag
+
+		if = {
+			limit = {
+				holder_scope = {
+					has_landed_title = c_neferset
+				}
+			}
+			save_event_target_as = target_dark_pharaoh
+		}
```

Comment:
`if` shouldn't be here. `holder_scope` must be used from title scope, not character. Plus, it's not needed since the event can fire only for the holder of c_neferset.